src/GafferArnoldUI/BarndoorVisualiser.cpp

```diff
@@ -0,0 +1,290 @@
+	if( member )
+	{
+		return member->readable();
+	}
+	return def;
+}
+
+const char *barndoorFragSource()
+{
+	return
+		"#version 150 compatibility\n"
+		""
+		""
+		"void main()"
+		"{"
+			"if( (int(gl_FragCoord.x) + int(gl_FragCoord.y)) % 2 == 0 )"
+			"{"
+				"discard;"
+			"}"
+			"else"
+			"{"
+				"gl_FragColor = vec4(0, 0, 0, 1);"
+			"}"
+		"}"
+		;
+}
+
+void addBarndoor( IECoreGL::GroupPtr group, BarndoorLocation location, float cornerLeft, float cornerRight )
+{
+
+	if( !( cornerLeft > 0 || cornerRight > 0 ) )
+	{
+		return;
+	}
+
+	cornerLeft = 1 - cornerLeft * 2.0;
+	cornerRight = 1 - cornerRight * 2.0;
+
+	IntVectorDataPtr vertsPerPoly = new IntVectorData;
+	std::vector<int> &vertsPerPolyVec = vertsPerPoly->writable();
+	vertsPerPolyVec.push_back( 4 );
+
+	IntVectorDataPtr vertIds = new IntVectorData;
+	std::vector<int> &vertIdsVec = vertIds->writable();
+	vertIdsVec.push_back( 0 );
+	vertIdsVec.push_back( 1 );
+	vertIdsVec.push_back( 2 );
+	vertIdsVec.push_back( 3 );
+
+	V3fVectorDataPtr p = new V3fVectorData;
+	std::vector<V3f> &pVec = p->writable();
+	pVec.push_back( V3f( -1, 1, 0  ) );
+	pVec.push_back( V3f( 1, 1, 0  ) );
+	pVec.push_back( V3f( 1, cornerRight, 0  ) );
+	pVec.push_back( V3f( -1, cornerLeft, 0  ) );
+
+	IECore::MeshPrimitivePtr mesh = new IECore::MeshPrimitive( vertsPerPoly, vertIds, "linear", p );
+
+	Imath::M44f trans;
+	switch( location )
+	{
+	case TOP:
+		break;
+	case BOTTOM:
+		trans.rotate( V3f( 0, 0, M_PI ) );
+		break;
+	case LEFT:
+		trans.rotate( V3f( 0, 0, M_PI / 2.0 ) );
+		break;
+	case RIGHT:
+		trans.rotate( V3f( 0, 0, M_PI / -2.0 ) );
+		break;
+	}
+
+	ToGLMeshConverterPtr meshConverter = new ToGLMeshConverter( mesh );
+
+	IECoreGL::GroupPtr barndoorGroup = new IECoreGL::Group();
+	barndoorGroup->getState()->add( new IECoreGL::Primitive::Selectable( false ) );
+	barndoorGroup->addChild( IECore::runTimeCast<IECoreGL::Renderable>( meshConverter->convert() ) );
+	barndoorGroup->setTransform( trans );
+
+	group->addChild( barndoorGroup );
+}
+
+} // namespace
+
+BarndoorVisualiser::BarndoorVisualiser()
+{
+}
+
+BarndoorVisualiser::~BarndoorVisualiser()
+{
+}
+
+IECoreGL::ConstRenderablePtr BarndoorVisualiser::visualise( const IECore::InternedString &attributeName, const IECore::ObjectVector *shaderVector, const IECore::Shader *lightShader, IECoreGL::ConstStatePtr &state ) const
+{
+	IECoreGL::ConstRenderablePtr standardResult = StandardLightFilterVisualiser::visualise( attributeName, shaderVector, lightShader, state );
+
+	if( !shaderVector || shaderVector->members().size() == 0 || !lightShader )
+	{
+		return standardResult;
+	}
+
+	const IECore::Shader *filterShader = IECore::runTimeCast<const IECore::Shader>( shaderVector->members().back().get() );
+	if( !filterShader )
+	{
+		return standardResult;
+	}
+
+	const IECore::CompoundData *filterShaderParameters = filterShader->parametersData();
+
+	float topLeft = parameterOrDefault( filterShaderParameters, "barndoor_top_left", 0.0f );
```

Comment:
What I also should've mentioned is that it's a little unfortunate that we're currently hard-coding the defaults to be used in absence of parameters. It probably doesn't happen so much that we actually have to use the defaults, but if it does it would be much nicer to use the data that was read by ArnoldShader already? Do you think there's any value in getting the defaults from there or should be keep it simple like this?